omod/src/main/webapp/resources/scripts/conditionlist/controllers/editcondition.controller.js

```diff
@@ -0,0 +1,129 @@
+        if($scope.condition.status == INACTIVE_STATUS && $scope.condition.onSetDate > $scope.condition.endDate){
+                emr.errorAlert("End Date can't be before Onset Date");
+        }
+        else{
+            var conditions = [];
+            conditions.push($scope.condition);
+            RestfulService.post('condition', conditions, function (data) {
+                emr.successAlert("Condition edited successfully")
+                window.location = '/' + OPENMRS_CONTEXT_PATH + '/coreapps/conditionlist/manageConditions.page?patientId=' + $scope.patientUuid + '&';
+            }, function (error) {
+                emr.errorAlert("Error Saving condition");
+            });
+        }
+    }
+
+    self.initCondition = self.initCondition || function () {
+            $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
+            $scope.condition = new ConditionModel($scope.patientUuid)
+            $scope.condition = CommonFunctions.extractUrlArgs(window.location.search)['condition'];
+            self.convertURLParameterToJSON();
+
+            $scope.concept = {
+                display: $scope.condition.concept.name,
+                concept: {
+                    display: $scope.condition.concept.name,
+                },
+                conceptName: {
+                    display: $scope.condition.concept.name,
+                }
+            }
+
+            var onsetDatePicker = angular.element(document.getElementsByName('conditionStartDate'))[0];     
+            onsetDatePicker.value =  new Date($scope.condition.onSetDate).toISOString().slice(0,10);
+          
+            if($scope.condition.status == INACTIVE_STATUS){
+                var endDatePicker = angular.element(document.getElementsByName('conditionEndDate'))[0];      
+                endDatePicker.value = new Date($scope.condition.endDate).toISOString().slice(0,10); 
+            } 
+
+            var inputs = document.getElementsByTagName('input');
+            inputs[0].disabled = true;
+            inputs[1].value = new Date($scope.condition.onSetDate).toISOString().slice(0,10);  
+            self.showEndDate();
+        }
+    
+    self.convertURLParameterToJSON = self.convertURLParameterToJSON || function(){    
+        $scope.condition = JSON.parse($scope.condition.split("%22").join("\"").split("%20").join(" "));
+    }
+        
+    self.showEndDate = self.showEndDate || function() {
+        var inputs = document.getElementsByTagName('input');
+        var groups = document.getElementsByClassName("group");
+        if($scope.condition.status == INACTIVE_STATUS){
+            groups[2].style.visibility = "visible";
+            inputs[3].value =new Date($scope.condition.endDate).toISOString().slice(0,10);
+        }
+        else{
+            groups[2].style.visibility = "hidden";    
+        }
+    }
+
+    self.validateCondition = self.validateCondition || function () {
+            var concept ;
```

Comment:
Shouldn't you assign the above to null?

Reply:
I will initialize it to null, but it will work even without assigning it to null as the values are being set in the if else code.